packages/ui/src/primitives/field.tsx

```diff
@@ -40,31 +48,73 @@ export const Label = React.forwardRef<HTMLLabelElement, React.HTMLAttributes<HTM
 
 export const Input = React.forwardRef(function Input(
   {
+    asChild,
     className,
     intent = 'idle',
+    state = 'native',
     ...props
-  }: React.InputHTMLAttributes<HTMLInputElement> & {
+  }: Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> & {
+    asChild?: boolean;
     intent?: FieldIntent;
+    type?: React.InputHTMLAttributes<HTMLInputElement>['type'] | 'otp';
+    state?: 'native' | 'hover' | 'focus-visible';
   },
   forwardedRef: React.ForwardedRef<HTMLInputElement>,
 ) {
+  const Comp = asChild ? Slot : 'input';
+
   return (
-    <input
+    <Comp
       data-field-input=''
       ref={forwardedRef}
       className={cn(
-        'block w-full bg-white text-gray-12 rounded-md bg-clip-padding py-1.5 px-2.5 border border-gray-a6 outline-none text-base',
-        'focus-visible:ring-[0.1875rem] disabled:opacity-50 disabled:cursor-not-allowed',
+        'flex w-full min-w-0 bg-white text-gray-12 rounded-md bg-clip-padding py-1.5 px-2.5 border outline-none text-base',
+        'disabled:opacity-50 disabled:cursor-not-allowed',
+        // type-specific overrides
+        props.type === 'otp'
+          ? 'min-h-10 text-[calc(var(--cl-font-size)*1.4)] font-semibold justify-center'
```

Comment:
Looks like these were 40x40 and centered. Should we keep them sized this way or keep them fluid like this?

![Screenshot 2024-06-07 at 9 58 51 AM](https://github.com/clerk/javascript/assets/825855/de48dc00-4fe7-4c2a-b3ab-d18c45e25b2b)

Reply:
This way deffo looks better!